src/bson_util.rs

```diff
@@ -158,12 +158,14 @@ fn num_decimal_digits(mut n: usize) -> usize {
 
 /// Read a document's raw BSON bytes from the provided reader.
 pub(crate) fn read_document_bytes<R: Read>(mut reader: R) -> Result<Vec<u8>> {
```

Comment:
These are the actual bugs found by fuzzing

Reply:
we _should_ be getting a warning when we use risky `as <numtype>` conversions as of #1045; I wonder why these ones weren't caught